framework/aster-frame/src/vm/page_table.rs

```diff
@@ -416,7 +416,12 @@ pub fn vaddr_to_paddr(vaddr: Vaddr) -> Option<Paddr> {
     // only to obtain the corresponding physical address according to the mapping.
     let last_entry = page_table.page_walk(vaddr, false)?;
     // FIXME: Support huge page
-    Some(last_entry.paddr() + (vaddr & (PAGE_SIZE - 1)))
+    Some(
+        last_entry
+            .paddr()
+            .checked_add(vaddr & (PAGE_SIZE - 1))
```

Comment:
Thanks for the observation, @tatetian. 

You are correct that since the physical address in the page table entry has its lowest `PAGE_SIZE_LOG2` bit cleared to zero. I will submit an update to fix this. Thanks again for your valuable feedback.